esp-hal/src/interrupt/riscv.rs

```diff
@@ -228,11 +229,53 @@ pub fn enable_direct(
     unsafe {
         map(Cpu::current(), interrupt, cpu_interrupt);
         set_priority(Cpu::current(), cpu_interrupt, level);
+
+        let mt = mtvec::read();
+
+        assert_eq!(
+            mt.trap_mode().into_usize(),
+            mtvec::TrapMode::Vectored.into_usize()
+        );
+
+        let base_addr = mt.address() as usize;
+
+        let int_slot = base_addr.wrapping_add((cpu_interrupt as usize) * 4);
+
+        let instr = encode_jal_x0(handler as usize, int_slot as usize)?;
```

Comment:
```suggestion
        let instr = encode_jal_x0(handler as usize, int_slot)?;
```
Nitpick, but `int_slot` is already a `usize`.